src/approaches/grammar_search_invention_approach.py

```diff
@@ -103,6 +116,43 @@ def __str__(self) -> str:
         return f"Forall[{type_sig}].[{str(self.body)}({objs})]"
 
 
+@dataclass(frozen=True, eq=False, repr=False)
+class _UnaryFreeForallClassifier(_UnaryClassifier):
+    """Universally quantify all but one variable in a multi-arity predicate.
+
+    Examples:
+        - ForAll ?x. On(?x, ?y)
+        - Forall ?y. On(?x, ?y)
+        - ForAll ?z. Between(?x, ?z, ?y)
+    """
+    body: Predicate  # Must be arity 2 or greater.
+    free_variable_idx: int
```

Comment:
If we're universally quantifying "all but one variable" then shouldn't this field actually be nonfree_variable_idx? That would make more sense since this classifier is unary.

Reply:
"free variable" refers to the variable that is exposed in the new predicate. free is the opposite of quantified